pkg/controller/backup_controller.go

```diff
@@ -1237,3 +1240,36 @@ func generateVolumeInfoFromDataUpload(backup *pkgbackup.Request, crClient kbclie
 
 	return tmpVolumeInfos
 }
+
+// generateVolumeInfoForLeftPV generates VolumeInfo for all the left
+// PVs (not skipped, no native snapshot, not CSI snapshot, no DataUpload, no PVB)
+func generateVolumeInfoForLeftPV(backup *pkgbackup.Request, volumeInfos []volume.VolumeInfo) []volume.VolumeInfo {
+	tmpVolumeInfos := make([]volume.VolumeInfo, 0)
+
+	volumeInfoMap := make(map[string]volume.VolumeInfo)
+	for _, volumeInfo := range volumeInfos {
+		volumeInfoMap[volumeInfo.PVName] = volumeInfo
+	}
+
+	for key, pvcPVInfo := range backup.PVMap {
+		// PVMap's key is either PVC name or PV name.
+		// PV name's format is pvcNamespace/pvName.
+		// This check only wants PV name key from PVMap.
+		if !strings.Contains(key, "/") {
+			if _, ok := volumeInfoMap[key]; !ok {
+				volumeInfo := volume.VolumeInfo{
+					PVCName:      pvcPVInfo.PVCName,
```

Comment:
these PVs should be marked as skipped in the finalized info list.

I'll double-check the remaining issues in https://github.com/vmware-tanzu/velero/issues/5834
Ideally there should not be any "left PV"

Reply:
Modified.